src/transport/SessionManager.cpp

```diff
@@ -656,6 +668,13 @@ void SessionManager::SecureUnicastMessageDispatch(const PacketHeader & packetHea
         return;
     }
 
+    // Drop secure unicast messages with privacy enabled.
+    if (packetHeader.HasPrivacyFlag())
```

Comment:
Shouldn't this happen up front before we DecodeAndConsume, so that does not get garbage data?  Similar to how unsecured messages do it.